src/aiodynamo/expressions.py

```diff
@@ -377,20 +377,24 @@ def encode(self, params: Parameters) -> str:
 
 class Condition(metaclass=abc.ABCMeta):
     def __and__(self, other: Condition) -> Condition:
-        if isinstance(self, AndCondition):
-            if isinstance(other, AndCondition):
-                return AndCondition(self.children.extending(other.children))
-            else:
-                return AndCondition(self.children.appending(other))
-        return AndCondition(MinLen2AppendOnlyList.create(self, other))
+        if isinstance(self, AndCondition) and isinstance(other, AndCondition):
+            return AndCondition(self.children + other.children)
+        elif isinstance(self, AndCondition):
+            return AndCondition(self.children + (other,))
+        elif isinstance(other, AndCondition):
```

Comment:
New and improved: #173 missed this case:
`A and (B and C)` 
That's when the user manually adds parens in non-intuitive way.

Reply:
nice, I'll add that to the other PR